src/vdaf/prio3.rs

```diff
@@ -620,6 +621,22 @@ impl<const SEED_SIZE: usize> Encode for Prio3PublicShare<SEED_SIZE> {
     }
 }
 
+impl<const SEED_SIZE: usize> PartialEq for Prio3PublicShare<SEED_SIZE> {
+    fn eq(&self, other: &Self) -> bool {
+        // Handle case that both join_rand_parts are populated.
+        if let Some(self_joint_rand_parts) = &self.joint_rand_parts {
```

Comment:
I think this would be easier to reason about if it was written as a match block on `(&self.joint_rand_parts, &other.joint_rand_parts)`.

Moreover, since the `(Some(_), Some(_))` branch is already using `ConstantTimeEq`, we could implement `ConstantTimeEq for Prio3PublicShare<SEED_SIZE>`, then have this `PartialEq` dispatch to it and convert the `Choice` to a `bool`.